### DISCUSSION TITLE
Queries with Aspire Redis + RedisInsight + SignalR

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#6816] [Q&A]
I'm struggling a bit with Redis + SingalR, I'm hoping to get some pointers.

**Issue 1: RedisInsight not deployed**

I'm declaring my redis cache in aspire via:

```cs
var cache = builder.AddRedis("cache").WithRedisInsight();
```

I see that container when running locally, but I don't see any additional container created for Redis Insight when provisioned to Azure. I can't see anything specifically created for it when I run 'azd infra synth', should I?

**Issue 2: Odd issue's with SignalR set to use the redis cache as a backplane for scale out.**

I've got a chat client using signalR, but I'm finding when deployed, I'm not getting group broadcast messages back. The connection looks fine in the browser and I can see it sending messages, but no responses coming back, usually just after deployment (which includes deploying both cache and the app).

SignalR is configured to use Redis:

```cs
builder.Services.AddSignalR(static options =>
{
	options.EnableDetailedErrors = true;
})
.AddStackExchangeRedis(builder.Configuration.GetConnectionString("cache")!, static options =>
{
	options.Configuration.ChannelPrefix = RedisChannel.Literal("chat");
});
```

The app with the signalR hub has sticky sessions enabled on ingress.

If I restart the cache container, it will start working again, so seems certainly related. I don't see any console errors, anything in the Aspire Dashboard (which is why I wanted RedisInsight, to see if there where any clues there). I can certainly look to Azure SignalR service, but I'd like to understand what's going on with the current setup, where this behavior comes from and how to get more information out of it.

**Comment by davidfowl** [comment]
> Issue 1: RedisInsight not deployed

RedisInsight, PgWeb, PgAdmin etc etc are only used in local dev. They are excluded in publish mode.

> Issue 2: Odd issue's with SignalR set to use the redis cache as a backplane for scale out.

Crank up the log level on the signalr logs.

**Reply by rithinch** [reply]
Hey @davidfowl, any specific reason for RedisInsight to be excluded from azure deployment?

What's the alternative workaround here to visualize and monitor the cache data from our application in the cloud?

Thanks.

**Reply by davidfowl** [reply]
Initially we were focused on these being inner loop focused only. Deploying these to production would be that we need to figure out the security model if you exposed them to the public internet. 

> What's the alternative workaround here to visualize and monitor the cache data from our application in the cloud?

On azure, application insights https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/application-insights

**Comment by kijanawoodard** [comment]
I just ran into this. Is it possible to override the publish mode restriction?

**Reply by davidfowl** [reply]
To expose your redis to the internet? No, not yet.

**Reply by kijanawoodard** [reply]
> To expose your redis to the internet? No, not yet.

Technically, I want "to be able to bring up redis insight in the dashboard". Not sure if that makes it any easier.

I've solved the "expose redis to internet" problem before by running a cloudflare tunnel container. Works well. I don't have that option in the current case.

**Reply by davidfowl** [reply]
No it does't make it easier. The dashboard does not have a secure tunnel to expose services.
